sway-lib-std/src/vec.sw

```diff
@@ -208,4 +208,32 @@ impl<T> Vec<T> {
         self.len -= 1;
         Option::Some(read(self.buf.ptr() + self.len * size_of::<T>()))
     }
+
+    /// Swaps two elements.
+    ///
+    /// # Arguments
+    ///
+    /// * element1_index - The index of the first element
+    /// * element2_index - The index of the second element
+    ///
+    /// # Reverts
+    ///
+    /// Reverts if `element1_index` or `element2_index` is greater than or equal to the length of vector.
+    pub fn swap(mut self, element1_index: u64, element2_index: u64) {
+        if element1_index == element2_index {
```

Comment:
I would actually move the `if` after the `asserts`. We do want the `swap` function to panic if the two indices are out-of-bounds even if they're equal.